arch/riscv/Kconfig

```diff
@@ -13,6 +13,14 @@ config COMPRESSED_ISA
 	bool
 	default y if 64BIT
 
+config FLOAT_HARD
+	bool "Enable hard-float calling convention"
+	default y
+	depends on FLOAT
+	select COMPRESSED_ISA
+	help
+	  This option enables the hard-float calling convention.
```

Comment:
As far as I can see, there is no ARM `softfp` ABI (i.e. using hardware FP instructions and registers with software floating-point ABI) equivalent in RISC-V.

I wonder if this symbol is really necessary since, as far as the current definition of the `FLOAT` symbol goes, it means hardware FP support is used ([it should really be called `CONFIG_FPU`](https://github.com/zephyrproject-rtos/zephyr/pull/23948#issuecomment-606565206)), in which case this symbol is always implied.

Reply:
There is a soft-floating point calling convention in RISC-V ([See section 18.3 of this document](https://riscv.org/wp-content/uploads/2015/01/riscv-calling.pdf)) and GCC does support enabling the F or D extensions while using the soft-float calling convention.

Reply:
After further investigation, I can see that there is indeed ARM `-mfloat-abi=softfp` equivalent in RISC-V:

> `-march=rv32imafdc -mabi=ilp32`: Hardware floating-point instructions can be generated, but no floating-point arguments will be passed in registers. This is like the `-mfloat-abi=softfp` argument to ARM’s GCC, and is usually used when interfacing with soft-float binaries on a hard-float system.

_(quoted from https://gnu-mcu-eclipse.github.io/toolchain/riscv/)_